lib/web_ui/lib/src/engine/semantics/label_and_value.dart

```diff
@@ -115,3 +98,44 @@ class LabelAndValue extends RoleManager {
     _cleanUpDom();
   }
 }
+
+String? computeDomSemanticsLabel({
+  String? tooltip,
+  String? label,
+  String? hint,
+  String? value,
+}) {
+  final String? labelHintValue = _computeLabelHintValue(label: label, hint: hint, value: value);
+
+  if (tooltip == null && labelHintValue == null) {
+    return null;
+  }
+
+  final StringBuffer combinedValue = StringBuffer();
+  if (tooltip != null) {
+    combinedValue.write(tooltip);
+
+    // Separate the tooltip from the rest via a line-break (if the rest exists).
+    if (labelHintValue != null) {
+      combinedValue.writeln();
+    }
+  }
+
+  if (labelHintValue != null) {
+    combinedValue.write(labelHintValue);
+  }
+
+  return combinedValue.isNotEmpty ? combinedValue.toString() : null;
```

Comment:
if `tooltip` is empty but not `null`, and `labelHintValue` is empty but not `null` this may end up returning just `\n`, right?

Does this need to `combinedValue.trim` before checking if it's empty?

Reply:
(labelHintValue cannot be empty-but-not-`null` because of how it's constructed in `_computeLabelHintValue` below)

Reply:
Good point, but I think it's already handled correctly. `labelHintValue` cannot be "empty but not null" because `_computeLabelHintValue` never returns an empty string. It is guaranteed to return either non-empty string or null. This is also captured by this this:

```dart
    expect(
      computeDomSemanticsLabel(tooltip: '', label: '', hint: '', value: ''),
      isNull,
    );
```

Unless I'm missing something.

Reply:
There's also extra protection from this expression:

```
    return computeDomSemanticsLabel(
      tooltip: semanticsObject.hasTooltip ? semanticsObject.tooltip : null,
      label: semanticsObject.hasLabel ? semanticsObject.label : null,
      hint: semanticsObject.hint,
      value: shouldDisplayValue ? semanticsObject.value : null,
    );
```

Reply:
Nah, it's just that I was reviewing top-to-bottom and didn't read the code for `_computeLabelHintValue` and the test until later :P